src/walletConnect/saga.ts

```diff
@@ -361,11 +423,37 @@ function* showActionRequest(request: Web3WalletTypes.EventArguments['session_req
 
   const supportedChains = yield* call(getSupportedChains)
 
+  const networkId = walletConnectChainIdToNetworkId[request.params.chainId]
+  const feeCurrencies = yield* select((state) => feeCurrenciesSelector(state, networkId))
+  let preparedTransactionsResult: PreparedTransactionsResult | undefined = undefined
+  if (
+    method === SupportedActions.eth_signTransaction ||
+    method === SupportedActions.eth_sendTransaction
+  ) {
+    const network = walletConnectChainIdToNetwork[request.params.chainId]
+    const normalizedTx = yield* call(
+      normalizeTransaction,
+      request.params.request.params[0],
+      network
+    )
+    preparedTransactionsResult = yield* call(prepareTransactions, {
+      feeCurrencies,
+      decreasedAmountGasFeeMultiplier: 1,
+      baseTransactions: [normalizedTx],
+    })
+  }
+
   navigate(Screens.WalletConnectRequest, {
     type: WalletConnectRequestType.Action,
     pendingAction: request,
     supportedChains,
     version: 2,
+    hasInsufficientGasFunds: preparedTransactionsResult?.type === 'not-enough-balance-for-gas',
+    feeCurrencies,
+    preparedTransaction:
+      preparedTransactionsResult?.type === 'possible'
+        ? getSerializablePreparedTransaction(preparedTransactionsResult.transactions[0])
```

Comment:
do we need to serialize `TransactionReqest` here only to [deserialize back to it later](https://github.com/valora-inc/wallet/blob/c60c69682f9e1b8dc26e99b9285517a587980f42/src/walletConnect/request.ts#L61)?

Reply:
it is mainly to prevent non serializable values being passed through navigation parameters. it generates a warning and the side effects are not so serious for us ([unserializable nav params mess with state persistence / deep linking](https://reactnavigation.org/docs/troubleshooting/#i-get-the-warning-non-serializable-values-were-found-in-the-navigation-state)) but i tried to avoid the bad practice here anyway. the prepared transaction then gets sent through a redux action, reducers also expect serializable values. non serializable action parameters stop Flipper from working ([context](https://valora-app.slack.com/archives/C025V1D6F3J/p1698416818574259))